File: VB/Step 3 - creating view 2 (PersonsViewModel and PersonsView)/Helpers/PersonViewModelCreator.vb
Imports System
Imports System.Collections.Generic
Imports System.Linq
Imports System.Text
Imports DX_MVVM.ViewModels

Namespace DX_MVVM.Helpers
    Public NotInheritable Class PersonViewModelCreator

        Private Sub New()
        End Sub

        Public Shared ReadOnly PersonViewModel() As PersonViewModel
        Public Shared ReadOnly Property DesignPersonViewModel() As PersonViewModel
            Get
                Return PersonViewModel(0)
            End Get
        End Property

        Shared Sub New()
            Dim length As Integer = PersonCreator.Person.Length
            PersonViewModel = New PersonViewModel(length - 1){}
            For i As Integer = 0 To length - 1
                PersonViewModel(i) = New PersonViewModel(PersonCreator.Person(i)) With {.IsReadOnly = True}
            Next i
        End Sub
    End Class
End Namespace

File: CS/Step 4 - building views/Helpers/PersonViewModelCreator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DX_MVVM.ViewModels;

namespace DX_MVVM.Helpers {
    public static class PersonViewModelCreator {
        public static readonly PersonViewModel[] PersonViewModel;
        public static PersonViewModel DesignPersonViewModel { get { return PersonViewModel[0]; } }

        static PersonViewModelCreator() {
            int length = PersonCreator.Person.Length;
            PersonViewModel = new PersonViewModel[length];
            for(int i = 0; i < length; i++)
                PersonViewModel[i] = new PersonViewModel(PersonCreator.Person[i]) { IsReadOnly = true };
        }
    }
}
